src/stream.jl

```diff
@@ -700,6 +700,10 @@ function callprocess(stream::TranscodingStream, inbuf::Buffer, outbuf::Buffer)
         makemargin!(outbuf, max(16, marginsize(outbuf) * 2))
     elseif state.code == :end && state.stop_on_end
         if stream.state.mode == :read
+            if stream.stream isa TranscodingStream && !has_sharedbuf(stream) && !iszero(buffersize(inbuf))
+                # unread data to match behavior if inbuf was shared.
+                GC.@preserve inbuf unsafe_unread(stream.stream, bufferptr(inbuf), buffersize(inbuf))
```

Comment:
Why do we not have a low-method that does the `GC.@preserve`?

Reply:
I guess the normal `unread` should be changed to work with any `AbstractVector{UInt8}`, then I could just `unread` a view of `inbuf.data`. Currently, it only supports `Union{Vector{UInt8},Base.CodeUnits{UInt8}}`